client.go

```diff
@@ -45,7 +61,7 @@ func newClient(baseURL, username, password string) (*Client, error) {
 
 // SetBaseURL sets the base URL for API requests to a custom endpoint. urlStr
 // should always be specified with a trailing slash.
-func (c *Client) SetBaseURL(urlStr string) error {
+func (c *RestClient) SetBaseURL(urlStr string) error {
```

Comment:
If `RestClient` contains method that are not included in the `Client` interface (like this one), maybe `NewClient` should still return a `*RestClient` instead of `Client`? Otherwise, how are you going to call `SetBaseURL` on a freshly instantiated client?

Reply:
Yes, you are right about that. The genereal idea is, that the interface contains every method an external user may ever need. For this specific case, we maybe could make this method unexported, since the only package-internal usage is in the `NewClient` function. What do you think about that?

Reply:
Now, that I think of it: `SetBaseURL` is currently _only_ called in the constructor function (which also accepts a base URL as parameter). It there any case _ever_ in which you'd need to change the base URL of an _existing_ client? (In which case, why not simply create a new client?)

Reply:
Yep, exactly! Gonna change it to unexported than. 👍